cmd/kcoin/chaincmd.go

```diff
@@ -126,20 +129,40 @@ Use "ethereum dump 0" to dump the genesis block.`,
 // the zero'd block (i.e. genesis) or will fail hard if it can't succeed.
 func initGenesis(ctx *cli.Context) error {
 	// Make sure we have a valid genesis JSON
+	genesis := new(core.Genesis)
```

Comment:
move this functionality to a  `knode/genesis/loader.go`?

the loader (or similar name) receives path, networkId and networkName, from these returns a genesis block or error